### PR TITLE
Update to `1.0.0` releases of `embedded-hal-*` packages

### PR DESCRIPTION
**Pull request by jessebraham**

### CODE REVIEW

**Comment by jessebraham** [comment]
The `1.0.0` release is now out, so once `embassy-time` has been updated I'll update this PR as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Updated to latest release candidate of `embedded-hal` (#1068)
```
Comment (MabezDev) [inline CHANGELOG.md:21]:
```suggestion
- Updated to `embedded-hal` v1.0 (#1068)
```

**Review by MabezDev — APPROVED** [review]
LGTM, feel free to merge once the embassy crates are released!

**Review by bjoernQ — APPROVED** [review]
LGTM

**Comment by jessebraham** [comment]
Just gonna merge this now to get it in `main` and avoid additional rebasing, I will remove the patches prior to release in a subsequent PR.

**Pull request merged**